umbraco/Umbraco-CMS
# V8: Add ability to resolve Health Checks with other property types - e.g. Content Picker

Link to existing issue: [#5452](https://github.com/umbraco/Umbraco-CMS/issues/5452)

**DESCRIPTION:**

This PR does a couple of things:

- Allows for using a new type of property editor (Content Picker) to resolve a Health Check
- Allows specifying a 'MissingErrorMessage' property within a Health Check that will display if a setting does not exist within a config file.

I've added the ability to use a Content Picker to resolve a Health Check because I want to add a Privacy Health Check that needs to verify that certain types of content items exist for a website. See gif below:

![healthcheck](https://user-images.githubusercontent.com/1780374/59356994-ed871180-8d21-11e9-840b-4b15d6d0c23a.gif)

If the setting that a Health Check is looking for initially does not exist in the config file it will explain to the user that the config setting is missing and needs to be added:

![config-setting-missing](https://user-images.githubusercontent.com/1780374/59357144-34750700-8d22-11e9-8c72-957c4d8d8745.JPG)

**TO TEST:**

To test you can verify against the Privacy Health Check feature that is planned to be released as part of the Our Umbraco HealthChecks package, once the Core changes in this PR are merged.

- Clone this PR & build the solution
- Clone Our.Umbraco.HealthChecks from: https://github.com/alanmac/Our.Umbraco.HealthChecks/tree/feature/privacy-health-check
- Reference the Umbraco.Web from this PR build within the Our.Umbraco.HealthChecks project - It needs to be aware of the Core changes made available with this PR.
- Reference the Our.Umbraco.HealthChecks dll output from the Core project so that the Health Checks are loaded for you to test.
- Copy App_Plugins/Our.Umbraco.HealthChecks from the Health Checks solution to the Umbraco Core solution so that the lang files are available.
- Build and Run

There should now be a Privacy Health Check available within the Health Check Dashboard.

Initially when you run the check it should tell you that required App Settings are missing. Once you add those and run the check again, it should allow you to use a Content Picker to point to existing content nodes (you'll need to have those created also) to resolve the issue.

Thanks to @lssweatherhead for her collaboration on this Core change.
